File: switch_statements/switch_case.py

def item1():
    a = int(input("Enter the value of a: "))
    b = int(input("Enter the value of b: "))
    op = input("Enter the operator")

    if op == "+":
        print(a+b)
    elif op == "-":
        print(a-b)
    elif op == "*":
        print(a*b)
    elif op == "/":
        print(a/b)
    elif op == "/":
         if b != 0:
            print(a / b)
         else:
            print("Division by zero is not allowed.")
    elif op == "%":
        print(a%b)
    else:
        print("Please Enter a valid Input Operation")


def counting():
    num = int(input("Enter the number till where you want to print:\n"))

    # count = " "
    for i in range(num + 1):
        print(i)
    

def isEven(num):
        return (num & 1) == 0
def EvenOrOdd():
     num = int(input("Enter the number which you want to check even or odd:\n"))

     if(isEven(num)):
         print(" Number is Even")
     else:
         print(" Number is Odd")


def factorial(num):
    fact = 1

    for i in range(num + 1):
        fact += i

    return fact

def nCr():
    n = int(input("Enter the value of n: "))
    r = int(input("Enter the value of r: "))

    num = factorial(n)
    denom = factorial(n) * factorial(n-r)

    print("The value of nCr is: ", num/denom)


def Power():
    num1 = int(input("Enter the base value: "))
    num2 = int(input("Enter the power value: "))

    answer = 1

    for i in range(num2):
        answer *= num1
    print("The value of Power is: ", answer)

    


def invalid_item():
     print("\nInvalid Number, Please select a number from the above given list :)\n")

def main():
    print("\nSelect the items from below given list which you want to print\n")
    print("1. Calculator")
    print("2. Counting")
    print("3. Check Even or Odd")
    print("4. Check nCr")
    print("5. Check Power")

    num_of_item = int(input("Enter the number: "))
    print()

    items = {
        1: item1,
        2: counting,
        3: EvenOrOdd,
        4: nCr,
        5: Power,
    }

    items.get(num_of_item, invalid_item)()

if __name__ == "__main__":
    main()

File: switch_statements/switch_case.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function item1(){
    rl.question("Enter the value of a: \n", (a) => {
        rl.question("Enter the value of b:\n", (b) => {
            rl.question("Enter the value of c:\n", (op) => {
                a = parseInt(a);
                b = parseInt(b);
                switch (op) {
                    case '+':
                        console.log(a+b)
                        break;
                    case '-':
                        console.log(a - b)
                        break;
                    case '*':
                        console.log(a * b)
                        break;
                    case '/':
                        if (b !== 0) {
                            console.log(a / b);
                        } else {
                            console.log("Division by zero is not allowed.");
                        }
                        break;
                    case '%':
                        console.log(a % b)
                        break;
                
                    default:
                        console.log("Please Enter a valid opeation.")
                        break;
                }
                rl.close();
            })
        })
    })
}

function counting(){
    rl.question("Enter the number till where you want to count: \n", (num) => {
        num = parseInt(num);
        console.log();

        for(let i = 0; i <= num; i++){
            console.log(i);
        }
        rl.close();
    })
}

function isEven(a) {
    return (a & 1) === 0;
}



function EvenOdd(){
    rl.question("Enter the number which you want to check: \n", (num) => {
        num = parseInt(num);
        console.log();

        if(isEven(num)){
            console.log(" Number is Even");
            } else {
            console.log(" Number is Odd");
        }
        rl.close();
    })
}

function factorial(num){
    let fact = 1;

    for(let i = 1; i <= num; i++){
        fact = fact * i;
    }

    return fact;
}

function nCr(){
    rl.question("Enter the number n: \n", (n) => {
        rl.question("Enter the number r: \n", (r) => {
            n = parseInt(n);
            r = parseInt(r);

            let num = factorial(n);
            let denom = factorial(r) * factorial(n - r);
            
            console.log("The value of NCR is : ", num/denom);
        })
    })
}


function Power(){
    rl.question("Enter the base number: \n", (num1) => {
        rl.question("Enter the poer number: \n", (num2) => {
            num1 = parseInt(num1);
            num2 = parseInt(num2);

            let answer = 1;

            for(let i = 1; i <= num2; i++){
                answer = answer * num1;
            }
            console.log("The value of Power is : ", answer);
        })
    })
}


function main(){
    console.log("\nSelect the items which you want to print from below list: \n");
    console.log();
    console.log("\n1. Calculator\n");
    console.log("\n2. Counting\n");
    console.log("\n3. Even or Odd\n");
    console.log("\n4. Check nCr\n");
    console.log("\n5. Check Power\n");

    rl.question("", (num_of_item)=> { 
        num_of_item = parseInt(num_of_item);
        console.log();
        
        switch (num_of_item) {
            case 1:
                item1();
                break;
            case 2:
                counting();
                break;
            case 3:
                EvenOdd();
                break;
            case 4:
                nCr();
                break;
            case 5:
                Power();
                break;
        
            default:
                console.log("\nInvalid Number, Please select a number from above given list :)\n");
                rl.close();
                break;
        }
    })
}

main();